docs/src/main/asciidoc/context-propagation.adoc

```diff
@@ -156,6 +156,114 @@ in order to get context propagated.
 The injected `ManagedExecutor` uses the Quarkus thread pool.
 ====
 
+== Overriding which contexts are propagated
+
+By default, all available contexts are propagated. However, you can override this behaviour in several ways.
+
+=== Using configuration
+
+The following configuration properties allow you to specify the default sets of propagated contexts:
+
+[cols="1,1,1"]
+|===
+|Configuration Key|Description|Default Value
+
+|`mp.context.ThreadContext.propagated`
+|The comma-separated set of propagated contexts
+|`Remaining` (all non-explicitly list contexts)
+
+|`mp.context.ThreadContext.cleared`
+|The comma-separated set of cleared contexts
+|`None` (no context), unless neither the propagated nor cleared sets contain `Remaining`, in which case the default is `Remaining` (all non-explicitly listed contexts)
+
+|`mp.context.ThreadContext.unchanged`
+|The comma-separated set of unchanged contexts
+|`None` (no context)
+|=== 
+
+The following contexts are available in Quarkus either out of the box, or depending on whether you include
+their extensions:
+
+[cols="1,1"]
+|===
+|Context Name|Description
+
+|`None`
+|Can be used to specify an empty set of contexts, but setting the value to empty works too
+
+|`Remaining`
+|All the contexts that are not explicitly listed in other sets
+
+|`Transaction`
+|The JTA transaction context
+
+|`CDI`
+|The CDI (ArC) context
+
+|`Servlet`
+|The servlet context
+
+|`JAX-RS`
+|The RESTEasy Classic context
+
+|`Application`
+|The current `ThreadContextClassLoader`
+|=== 
+
+=== Overriding the propagated contexts using annotations
+
+In order for automatic context propagation, such as Mutiny uses, to be overridden in specific methods,
+you can use the https://javadoc.io/doc/io.smallrye/smallrye-context-propagation-api/latest/io/smallrye/context/api/CurrentThreadContext.html[`@CurrentThreadContext`]
+annotation:
+
+[source,java]
+----
+    // Get the prices stream
+    @Inject
+    @Channel("prices") Publisher<Double> prices;
+
+    @GET
+    @Path("/prices")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    @SseElementType(MediaType.TEXT_PLAIN)
+    // Get rid of all context propagation, since we don't need it here
+    @CurrentThreadContext(unchanged = ThreadContext.ALL_REMAINING)
```

Comment:
For sake of clarity in the example it might be best to explicily declare the other two attributes as empty arrays?

When I read it I was left wondering "all remaining ... compared to what since I didn't declare anything at all?"

Reply:
Well, they're default values. I can add them, but that's really more verbose, do you really think it's worth repeating the default values? `Remaining` is explained in the docs, though perhaps I should mention the `ThreadContext` constant that refers to it, now that you mention it.

Reply:
Added this to the docs.